reaktive/src/commonMain/kotlin/com/badoo/reaktive/single/BlockingGet.kt

```diff
@@ -0,0 +1,58 @@
+package com.badoo.reaktive.single
+
+import com.badoo.reaktive.disposable.Disposable
+import com.badoo.reaktive.utils.Lock
+import com.badoo.reaktive.utils.atomicreference.AtomicReference
+import com.badoo.reaktive.utils.synchronized
+
+fun <T> Single<T>.blockingGet(): T {
+    val lock = Lock()
```

Comment:
If we still have locks, why you don't use same the same way in was before in some dangerous for race condition places?

Reply:
See comments above. Here we need `Lock` as it how `Condition` works.